src/functional/ffront/past_to_itir.py

```diff
@@ -107,7 +107,14 @@ def visit_Program(
         #  a workaround we add additional arguments to the fencil definition
         #  containing the size of all fields. The caller of a program is (e.g.
         #  program decorator) is required to pass these arguments.
-        size_params = self._gen_size_params_from_program(node)
+
+        params = [itir.Sym(id=inp.id) for inp in node.params]
+        params_add = False
+        for _i, body_entry in enumerate(node.body):
+            if "domain" not in body_entry.kwargs:
+                params_add = True
+
+        params = params + self._gen_size_params_from_program(node) if params_add else params
```

Comment:
I think the following is easier to read:
```suggestion
        if params_add:
            params = params + self._gen_size_params_from_program(node)
```